### PR TITLE
fold in python cookbooks to travis_build_environment

### PR DESCRIPTION
**Pull request by Lyoness**

### CODE REVIEW

File: cookbooks/travis_python/recipes/system.rb [hunk 1]
```diff
@@ -1,76 +0,0 @@
...
-
-  execute "install packages in #{venv_name}" do
-    command "#{venv_fullname}/bin/pip install --upgrade #{packages.join(' ')}"
-    user node['travis_build_environment']['user']
-    group node['travis_build_environment']['group']
-    environment(
-      'HOME' => node['travis_build_environment']['home']
-    )
-  end
-end
```
Comment (meatballhat) [inline cookbooks/travis_python/recipes/system.rb]: Copied over as `system_python.rb`, that is 😃

Comment (Lyoness) [inline cookbooks/travis_python/recipes/system.rb]: doh. :+1:

File: cookbooks/travis_build_environment/recipes/pyenv.rb [hunk 2]
```diff
@@ -1,29 +1,6 @@
...
-# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-# THE SOFTWARE.
-
 virtualenv_root = "#{node['travis_build_environment']['home']}/virtualenv"
 
-include_recipe 'travis_python::virtualenv'
+include_recipe 'travis_build_environment::virtualenv'
 
 package %w(
   build-essential
```
Comment (meatballhat) [inline cookbooks/travis_build_environment/recipes/pyenv.rb]: I'm pretty sure this recipe is no longer necessary.  These packages and the pyenv git repo are handled in the `travis_build_environment::python` recipe, as is the inclusion of `travis_build_environment::virtualenv`.

File: cookbooks/travis_build_environment/recipes/python.rb [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    user node['travis_build_environment']['user']
+    group node['travis_build_environment']['group']
+    environment(
+      'HOME' => node['travis_build_environment']['home']
+    )
+  end
+end
+
+template '/etc/profile.d/pyenv.sh' do
+    source 'pyenv.sh.erb'
```
Comment (meatballhat) [inline cookbooks/travis_build_environment/recipes/python.rb]: looks like an extra level of indentation on this line

File: cookbooks/travis_build_environment/recipes/python.rb [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    group node['travis_build_environment']['group']
+  end
+
+  bindirs << "/opt/python/#{py}/bin"
+
+  bash "create virtualenv at #{venv_fullname} from #{py}" do
+  code "virtualenv --python=/opt/python/#{py}/bin/python #{venv_fullname}"
+  owner node['travis_build_environment']['user']
+  group node['travis_build_environment']['group']
+end
```
Comment (meatballhat) [inline cookbooks/travis_build_environment/recipes/python.rb]: Indent these four lines 😁

File: cookbooks/travis_build_environment/recipes/virtualenv.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+
+
+include_recipe 'travis_build_environment::pip'
+
+execute 'pip install virtualenv==15.1.0' do
+  version '15.1.0'
+  action :install
+end
```
Comment (meatballhat) [inline cookbooks/travis_build_environment/recipes/virtualenv.rb]: These three lines inside the block are not valid for the `execute` resource, so they can be removed.

File: cookbooks/travis_build_environment/recipes/python.rb (hunk 4 above)
Comment (Lyoness) [inline cookbooks/travis_build_environment/recipes/python.rb]: rubocop beat you to it.

File: cookbooks/travis_build_environment/recipes/python.rb [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    owner node['travis_build_environment']['user']
+    group node['travis_build_environment']['group']
+  end
+
+  bindirs << "/opt/python/#{py}/bin"
+
+  bash "create virtualenv at #{venv_fullname} from #{py}" do
+  code "virtualenv --python=/opt/python/#{py}/bin/python #{venv_fullname}"
+  owner node['travis_build_environment']['user']
+  group node['travis_build_environment']['group']
```
Comment (meatballhat) [inline cookbooks/travis_build_environment/recipes/python.rb]: So close :grin:  The body of the block should be indented.  This should be auto-correctable via `rubocop -a --auto-gen-config`, fwiw.

File: cookbooks/travis_build_environment/recipes/python.rb [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    )
+  end
+end
+
+template '/etc/profile.d/pyenv.sh' do
+  source 'pyenv.sh.erb'
+    owner node['travis_build_environment']['user']
+    group node['travis_build_environment']['group']
+    mode 0o644
+    variables(
```
Comment (meatballhat) [inline cookbooks/travis_build_environment/recipes/python.rb]: So close again.  The `source`, `owner`, `group`, `mode`, `variables`, and `backup` calls should all be aligned one indentation level in from the outer `template` resource.  Rubocop auto fix should handle this for you ❤️

**Pull request merged**